presto-main/src/main/java/com/facebook/presto/operator/scalar/ScaleFunction.java

```diff
@@ -0,0 +1,239 @@
+/*
+ * Licensed to Ted Dunning under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.operator.scalar;
+
+/**
+ * Encodes the various scale functions for t-digests. These limits trade accuracy near the tails against accuracy near
+ * the median in different ways. For instance, K_0 has uniform cluster sizes and results in constant accuracy (in terms
+ * of q) while K_3 has cluster sizes proportional to min(q,1-q) which results in very much smaller error near the tails
+ * and modestly increased error near the median.
+ * <p>
+ * The base forms (K_0, K_1, K_2 and K_3) all result in t-digests limited to a number of clusters equal to the
+ * compression factor. The K_2_NO_NORM and K_3_NO_NORM versions result in the cluster count increasing roughly with
+ * log(n).
+ */
+public enum ScaleFunction
+{
+    /**
+     * Generates uniform cluster sizes. Used for comparison only.
+     */
+    K_0 {
```

Comment:
Looks unused?

Reply:
It is unused, but I left it for testing purposes because K_0 gives a linear scaling function which is not biased at the tails.